packages/api/endpoints/providers.js

```diff
@@ -129,22 +172,26 @@ async function put({ params: { id }, body }, res) {
  */
 async function del(req, res) {
   const providerModel = new Provider();
+  const knex = await getKnexClient({ env: process.env });
 
   try {
-    await providerModel.delete({ id: req.params.id });
-
-    if (inTestMode()) {
-      const esClient = await Search.es(process.env.ES_HOST);
-      await esClient.delete({
-        id: req.params.id,
-        type: 'provider',
-        index: process.env.ES_INDEX,
-      }, { ignore: [404] });
-    }
+    await knex.transaction(async (trx) => {
+      await trx(tableNames.providers).where({ name: req.params.id }).del();
+      await providerModel.delete({ id: req.params.id });
+      if (inTestMode()) {
+        const esClient = await Search.es(process.env.ES_HOST);
+        await esClient.delete({
+          id: req.params.id,
+          type: 'provider',
+          index: process.env.ES_INDEX,
+        }, { ignore: [404] });
+      }
+    });
     return res.send({ message: 'Record deleted' });
   } catch (error) {
-    if (error instanceof AssociatedRulesError) {
-      const message = `Cannot delete provider with associated rules: ${error.rules.join(', ')}`;
+    if (error instanceof AssociatedRulesError || error.constraint === 'rules_providercumulusid_foreign') {
```

Comment:
I'm not entirely happy with this error catch - it's relying on the table constraint name, which is subject to change via migrations. 

I've added a unit test around the desired behavior, and the units all *run* the migrations, so the logic is covered, but I'm open to suggestions on a more pleasant way to sort this.